### PR TITLE
Improve quoted-printable decoding

### PR DESCRIPTION
**Pull request by Maria-12648430**
This PR improves Quoted-Printable decoding by removing look-aheads in the remaining binary, and processing the entire body in one go instead of line by line.

This PR also includes a small (unrelated) nano-optimization/correction, replacing a list with a tuple in a case expression and the associated matches.

### CODE REVIEW

**Comment by mworrell** [comment]
Hi @Maria-12648430, we merged a re-format of the code, which makes your PR conflict with master 😭 

Could you rebase?

**Comment by Maria-12648430** [comment]
> Could you rebase?

It will probably have to wait until next week, but yes, sure 😊

**Comment by Maria-12648430** [comment]
Well, I did it today ;) For reference, I rebased my branch on `master`, solved the merge conflicts, then ran `rebar3 fmt`. Hope it all turned out as you expect.

File: src/mimemail.erl [hunk 1]
```diff
@@ -2429,8 +2398,10 @@ rfc2047_decode_test_() ->
             ?assertEqual(
                 <<"ab">>,
                 decode_header(
-                    <<"=?ISO-8859-1?Q?a?=\n"
-                    "		=?ISO-8859-1?Q?b?=">>,
+                    <<
+                        "=?ISO-8859-1?Q?a?=\n"
+                        "		=?ISO-8859-1?Q?b?="
+                    >>,
```
Comment (Maria-12648430) [inline src/mimemail.erl:2404]: This was not part of my original PR, it got reformatted with the rest when I ran `rebar3 fmt`.

**Review by seriyps — APPROVED** [review]
Thanks! Can't say I 100% understand QP-encoding (especially soft-linebreaks and whitespace management).
But since code is now simpler and more optimized + all tests pass - I'm for it.

**Comment by mworrell** [comment]
Thanks @Maria-12648430 and sorry for the needed rebase.

**Comment by Maria-12648430** [comment]
> Thanks @Maria-12648430 and sorry for the needed rebase.

No worries :smile:

**Pull request merged**